frontend/src/stores/socketStore.ts

```diff
@@ -0,0 +1,63 @@
+import io from 'socket.io-client';
+import { defineStore } from 'pinia';
+import type { SocketStore } from '@/types/Sockets';
+import { useChatStore } from './chatStore';
+import { useMessageStore } from './messageStore';
+import type { Chat_List_Item, SingleMessage } from '@/types/Chat';
+import type { SocketMessage } from '@/types/Sockets';
+
+// it's always only one user..
+
+export const useSocketStore = defineStore('sockets', {
+	state: (): SocketStore => ({
+		chats: {
+			socket: null,
+			initialized: false,
+			in_rooms: [],
+		},
+	}),
+	actions: {
+		initialize() {
+			if (this.chats.initialized === false) {
+				this.initializeChats();
+			}
+		},
+		async initializeChats() {
+			this.chats.initialized = true;
+			// get initial data in chat store if it is not there
+			await useChatStore().init(false);
+			// create socket
+			this.chats.socket = io('/chats', { withCredentials: true });
+			this.chats.socket.on('connection', (conn) => {
+				console.log('Connected: ', conn);
+			});
+			this.chats.socket.on(
+				'chatListUpdate',
+				(update: SocketMessage<Chat_List_Item>) => {
+					useChatStore().socketAction(update);
+				},
+			);
+			this.chats.socket.on(
+				'messageListUpdate',
+				(update: SocketMessage<SingleMessage>) => {
+					useMessageStore().socketAction(update);
+				},
+			);
+		},
+		disconnect() {
+			this.chats.socket?.off('newMessage');
```

Comment:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining

> The optional chaining (?.) operator accesses an object's property or calls a function. If the object accessed or function called using this operator is undefined or null, the expression short circuits and evaluates to undefined instead of throwing an error.